explorer/explorerroutes.go

```diff
@@ -235,6 +235,54 @@ func (exp *explorerUI) Mempool(w http.ResponseWriter, r *http.Request) {
 	io.WriteString(w, str)
 }
 
+// Ticketpool is the page handler for the "/ticketpool" path
+func (exp *explorerUI) Ticketpool(w http.ResponseWriter, r *http.Request) {
+	chartData, groupedTickets, err := exp.explorerSource.TicketPoolVisualization(
+		dbtypes.ChartGroupingFromStr("all"),
+	)
+	if err != nil {
+		log.Errorf("Template execute failure: %v", err)
+		exp.StatusPage(w, defaultErrorCode, defaultErrorMessage, ErrorStatusType)
+		return
+	}
+	var mp = dbtypes.PoolTicketsData{}
+	exp.MempoolData.Lock()
+	var mpData = exp.MempoolData
+	exp.MempoolData.Unlock()
```

Comment:
You just copied the `MempoolData` pointer at this point, so you can't unlock yet.  Down below you actually read the data pointed to by `MempoolData` (`mpData`), so you'd need to unlock after that to prevent a writer concurrently modifying this data.

Also, since you are not modifying the data, you just need `RLock` and `RUnlock`, which will not lock out other readers, only writers.

Reply:
ok. Noted